models/perm/access/repo_permission.go

```diff
@@ -253,6 +254,34 @@ func finalProcessRepoUnitPermission(user *user_model.User, perm *Permission) {
 	}
 }
 
+// GetActionsUserRepoPermission returns the actions user permissions to the repository
+func GetActionsUserRepoPermission(ctx context.Context, repo *repo_model.Repository, actionsUser *user_model.User, taskID int64) (perm Permission, err error) {
+	if actionsUser.ID != user_model.ActionsUserID {
+		setting.PanicInDevOrTesting("GetActionsUserRepoPermission can only be called by the actions user")
```

Comment:
I think we need a hard panic here, to prevent the code from executing.

Reply:
Every caller currently ensure this panic is always skipped, but I wanted to add a bomb for unit tests. (previously no tests)

Not sure if always panic / error is better
```suggestion
		return perm, fmt.Errorf("api GetActionsUserRepoPermission can only be called by the actions user")
```
both should make a http 500, I guess. (somewhere some go linter told ne to not capitalize errors at the beginning)

---

calling  NewActionUser() would work for me as well, but I thought about avoiding this allocation and pass the previously checked doer

Reply:
Since `GetActionsUserRepoPermission` already uses `error` return value, so I think error is better.